File: sdks/java/src/main/java/com/puxbay/resources/GiftCardsResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.GiftCard;
import java.util.HashMap;
import java.util.Map;

public class GiftCardsResource {
    private final Puxbay client;

    public GiftCardsResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<GiftCard> list(int page, String status) throws PuxbayException {
        String query = "gift-cards/?page=" + page;
        if (status != null) {
            query += "&status=" + status;
        }
        return (PaginatedResponse<GiftCard>) client.request("GET", query, null, PaginatedResponse.class);
    }

    public GiftCard get(String cardId) throws PuxbayException {
        return client.request("GET", "gift-cards/" + cardId + "/", null, GiftCard.class);
    }

    public GiftCard create(GiftCard card) throws PuxbayException {
        return client.request("POST", "gift-cards/", card, GiftCard.class);
    }
    
    public GiftCard redeem(String cardId, double amount) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("amount", amount);
        return client.request("POST", "gift-cards/" + cardId + "/redeem/", body, GiftCard.class);
    }
    
    public Map checkBalance(String code) throws PuxbayException {
        return client.request("GET", "gift-cards/check-balance/?code=" + code, null, Map.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/GiftCardsResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.GiftCard
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class GiftCardsResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, status: String? = null): PaginatedResponse<GiftCard> {
        val statusParam = if (status != null) "&status=$status" else ""
        return client.request(HttpMethod.Get, "gift-cards/?page=$page$statusParam")
    }
    
    suspend fun get(cardId: String): GiftCard =
        client.request(HttpMethod.Get, "gift-cards/$cardId/")
    
    suspend fun create(card: GiftCard): GiftCard =
        client.request(HttpMethod.Post, "gift-cards/", card)
    
    suspend fun redeem(cardId: String, amount: Double): GiftCard =
        client.request(HttpMethod.Post, "gift-cards/$cardId/redeem/", mapOf("amount" to amount))
    
    suspend fun checkBalance(code: String): Map<String, Any> =
        client.request(HttpMethod.Get, "gift-cards/check-balance/?code=$code")
}
